python/dgl/sparse/elementwise_op_diag.py

```diff
@@ -49,20 +49,22 @@ def diag_add(
     return NotImplemented
 
 
-def diag_sub(D1: DiagMatrix, D2: DiagMatrix) -> DiagMatrix:
+def diag_sub(
+    D1: DiagMatrix, D2: Union[DiagMatrix, SparseMatrix]
+) -> Union[DiagMatrix, SparseMatrix]:
     """Elementwise subtraction
 
     Parameters
     ----------
     D1 : DiagMatrix
         Diagonal matrix
-    D2 : DiagMatrix
-        Diagonal matrix
+    D2 : DiagMatrix or SparseMatrix
+        Diagonal matrix or sparse matrix
 
     Returns
     -------
     DiagMatrix
-        Diagonal matrix
+        Diagonal matrix, same as D2
```

Comment:
the result of D - A should be sparse, right?

Reply:
fixed